examples/s2_field_combination_extern_rf_train/nrw_crop_extern_s2_workflow_to_datasets_with_timeshift.ipynb

```diff
@@ -0,0 +1,2364 @@
+       "      {'type': 'TimeShift',\n",
+       "       'params': {'type': 'relative', 'granularity': 'months', 'value': -2},\n",
+       "       'sources': {'source': {'type': 'GdalSource',\n",
+       "         'params': {'data': {'type': 'internal',\n",
+       "           'datasetId': '47d9ee89-6d87-4228-824d-46c52c751db0'}}}}},\n",
+       "      {'type': 'TimeShift',\n",
+       "       'params': {'type': 'relative', 'granularity': 'months', 'value': -2},\n",
+       "       'sources': {'source': {'type': 'GdalSource',\n",
+       "         'params': {'data': {'type': 'internal',\n",
+       "           'datasetId': '3db91688-19f2-4dca-9995-0f4097b3996a'}}}}},\n",
+       "      {'type': 'TimeShift',\n",
+       "       'params': {'type': 'relative', 'granularity': 'months', 'value': -2},\n",
+       "       'sources': {'source': {'type': 'GdalSource',\n",
+       "         'params': {'data': {'type': 'internal',\n",
+       "           'datasetId': '2a36bca3-fbe9-4d24-854f-fa159b755bfb'}}}}}]}},\n",
+       "   'rasters': [{'type': 'TimeShift',\n",
+       "     'params': {'type': 'relative', 'granularity': 'months', 'value': -1},\n",
+       "     'sources': {'source': {'type': 'GdalSource',\n",
+       "       'params': {'data': {'type': 'internal',\n",
+       "         'datasetId': 'cf14cbe6-d747-48ca-abfe-bd9ea87618a4'}}}}},\n",
+       "    {'type': 'TimeShift',\n",
+       "     'params': {'type': 'relative', 'granularity': 'months', 'value': -1},\n",
+       "     'sources': {'source': {'type': 'GdalSource',\n",
+       "       'params': {'data': {'type': 'internal',\n",
+       "         'datasetId': 'a22225d9-1e96-4646-ad67-0efe41291361'}}}}},\n",
+       "    {'type': 'TimeShift',\n",
+       "     'params': {'type': 'relative', 'granularity': 'months', 'value': -1},\n",
+       "     'sources': {'source': {'type': 'GdalSource',\n",
+       "       'params': {'data': {'type': 'internal',\n",
+       "         'datasetId': '47d9ee89-6d87-4228-824d-46c52c751db0'}}}}},\n",
+       "    {'type': 'TimeShift',\n",
+       "     'params': {'type': 'relative', 'granularity': 'months', 'value': -1},\n",
+       "     'sources': {'source': {'type': 'GdalSource',\n",
+       "       'params': {'data': {'type': 'internal',\n",
+       "         'datasetId': '3db91688-19f2-4dca-9995-0f4097b3996a'}}}}},\n",
+       "    {'type': 'TimeShift',\n",
+       "     'params': {'type': 'relative', 'granularity': 'months', 'value': -1},\n",
+       "     'sources': {'source': {'type': 'GdalSource',\n",
+       "       'params': {'data': {'type': 'internal',\n",
+       "         'datasetId': '2a36bca3-fbe9-4d24-854f-fa159b755bfb'}}}}}]}}}"
+      ]
+     },
+     "execution_count": 34,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "points_with_s2_cloud_free_shift = points_with_s2_cloud_free\n",
+    "\n",
+    "for month_delta in range(-8,0):\n",
```

Comment:
I don't understand the loop here. Wouldn't you just need to loop inside sources?

Maybe you could also unroll the loop to make it more clear?

Reply:
the RF wants this input:

`B02, B03, B04, ...., B02-1m, B03-1m, B04-1m, .... B02-8m, B03-8m, B04-8m....`

So i need to shift the sources by the months in the range AND then i need to join all the results to the points. The RasterVectorJoin throws an Error when you use more then 8 Inputs.